### ISSUE TITLE
`BundleTaskExtension$BuildAction` adds undeclared inputs to `jar` tasks

### ISSUE THREAD

**Issue by clayburn** [bndtools/bnd#5279]
Discovered in: https://github.com/ben-manes/caffeine/issues/728. The `BuildAction` class in the `BundleTaskExtension` is affecting the output of the `jar` task by changing the manifest based on certain properties but not declaring them as inputs. 

This was noticed in https://github.com/ben-manes/caffeine/issues/728 regarding the `archiveVersion` of the `jar` task. In this project, the `jar` task was made cacheable. Before that, the task would be considered out of date in the below scenario, since the name of the output file had changed. But when the `jar` task is made cacheable, the output file name is not considered as part of the cache key, so an invalid cache hit results in a jar without the newer version.

`archiveVersion` is just one example in this action. It appears there are other properties in this action that are effectively inputs but are not declared.

To reproduce:

1. Checkout caffeine @ 340e9840. 
2. `./gradlew clean :caffeine:jar`
3. `unzip -p caffeine/build/libs/caffeine-3.1.2-SNAPSHOT.jar META-INF/MANIFEST.MF | grep Bundle-Version`
4. Verify the `Bundle-Version` is `3.1.2.SNAPSHOT`
5. `./gradlew :caffeine:jar -Prelease`
6. `unzip -p caffeine/build/libs/caffeine-3.1.2.jar META-INF/MANIFEST.MF | grep Bundle-Version`
7. Verify the `Bundle-Version` is still `3.1.2.SNAPSHOT`, when the desired outcome is `3.1.2`

**Comment by bjhargrave** [comment]
@clayburn Thanks for the bug report.

I have a fix PR #5280 for Bnd 6.4. In the meantime, here is a workaround for your build that will work with Bnd 6.3:

```diff
diff --git a/caffeine/build.gradle b/caffeine/build.gradle
index 260e0e4e..a30284e1 100644
--- a/caffeine/build.gradle
+++ b/caffeine/build.gradle
@@ -87,6 +87,9 @@ tasks.named('jar').configure {
   dependsOn compileCodeGenJava
   manifest {
     attributes 'Bundle-SymbolicName': 'com.github.ben-manes.caffeine'
+    attributes 'Bundle-Version': aQute.bnd.version.MavenVersion.parseMavenString(project.version.toString())
+        .getOSGiVersion()
+        .toString()
     attributes 'Import-Package': ''
     attributes 'Export-Package': [
       'com.github.benmanes.caffeine',
```

**Comment by clayburn** [comment]
Awesome, thanks for the quick fix!
